tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1316,4 +1309,160 @@ def test_bad_attr(dut):
         assert False, "Expected AttributeError"
 
 
-from test_cocotb_35 import *
```

Comment:
I'm almost wondering if this is worth keeping separate, but renaming to `test_async_await` - this file is already pretty long.

Reply:
Splitting the file up is a good idea, but it should probably be done on a feature basis, not syntax. Also, we probably should be using `MODULE` instead of importing tests like this. But this is out of scope for this PR.

Reply:
Generally good points. The only danger of merging now is that splitting again later makes the history marginally hard to follow. Your commits are nicely separated anyway though, so it shouldn't be too hard to follow even if we do split that file up again later.